tools/tagger/index.html

```diff
@@ -400,11 +511,89 @@
     let lang=params.get('lang');
     if (!lang) lang='en';
 
+    const asyncForEach = async (array, callback) => {
+      for (let index = 0; index < array.length; index += 1) {
+        await callback(array[index], index, array);
+      }
+    };
+
+    const checkLinks = params.get('checkLinks') || false;
+
+    let lcRunning = false;
+    const runLinkchecker = async () => {
+      if (lcRunning) return;
+      lcRunning = true;
+
+      document.getElementById('linkchecker').classList.remove('hidden');
+
+      const LOADING_ICON = '<i class="gg-spinner"></i>';
+      const OK_ICON = '<i class="gg-check-o"></i>';
+      const ERROR_ICON = '<i class="gg-danger"></i>';
+
+      window.blog = {
+        language: lang,
+        context: '/'
+      };
+      
+      const { getLink } = await import('/scripts/common.js');
+
+      const computeLink = (tagInfo) => {
+        const { link } = tagInfo;
+        if (link) {
+          // adapt host to current browser host
+          const u = new URL(link);
+          const current = new URL(window.location.href);
+          return `${current.origin}${u.pathname}`;
+        } else {
+          return getLink('topic', tagInfo.tag.replace(/\s/gm, '-').toLowerCase());
+        }
+      };
+      
+      const resultsPanel = document.getElementById('lcResults');
+      await asyncForEach(CATEGORIES, async (cat, index) => {
+        await asyncForEach(document.querySelectorAll(`#tags-data #cat-${index} li`), async (e) => {
+          const tagInfo = fetchTag(e.id);
+          if (tagInfo) {
+            const { tag, nuft } = tagInfo;
+            const link = computeLink(tagInfo);
+
+            if (link) {
+              console.log(`checking link ${tag}: ${link}`, tagInfo);
+              const ui = document.querySelector(`#results [data-target="${e.id}"]`);
+              const icon = document.createElement('span');
+              icon.classList = 'lcIcon';
+              icon.innerHTML = LOADING_ICON;
+              icon.title = `Checking if ${link} exists.`
+              ui.append(icon);
+              icon.addEventListener('click', () => {
+                window.open(link);
+              });
+
+              const res = await fetch(link);
```

Comment:
Would using a `HEAD` request make it a bit faster?

Reply:
Good idea!

Reply:
Interestingly, the `HEAD` requests seem to be much slower. I assume the `GET` might be cached somewhere from a previous hit while the `HEAD` request will always be new.

Reply:
Reverted to `GET` requests since definitively faster.